src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
+        if (perf == null) {
+            perf = new Perfection();
+        }
+        return perf;
+    }
+
+
+    public static boolean isPerfect(long candidate) {
+        boolean retVal;
+        long[] divisors = GetDivisors(candidate);
+        int sum = 0;
+        for (int d = 0; d < 1000; d++)
+        {
+            sum = sum + divisors[d];
+        }
+        if (sum == candidate)
+            retVal = true;
+        return retVal;
```

Comment:
This method has some issues:
- `retVal` is not initialized.
- `sum` is declared as `int` and should be a `long` to make the sum.
- the loop over the `divisors` is using a hard coded 1000 as limit, that will fail in smaller arrays.
- no need to compare the sum, you can just return it :) 

Lets simplify a little bit:
```suggestion
    Long[] divisors = getDivisors(candidate);
    long sum = 0;

    for (long divisor : divisors) {
      sum += divisor;
    }

    return sum == candidate;
```